codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/extensions/DafnyPythonLocalServiceStructureGenerator.java

```diff
@@ -45,17 +46,122 @@ public DafnyPythonLocalServiceStructureGenerator(
 
   @Override
   public void run() {
+    Set<ShapeId> localServiceConfigShapes =
+      SmithyNameResolver.getLocalServiceConfigShapes(model);
+
     if (shape.hasTrait(PositionalTrait.class)) {
       // Do not need to render shapes with positional trait, their linked shapes are rendered
       return;
-    }
-    if (!shape.hasTrait(ErrorTrait.class)) {
+    } else if (localServiceConfigShapes.contains(shape.getId())) {
+      renderLocalServiceConfigShape();
+    } else if (!shape.hasTrait(ErrorTrait.class)) {
       renderStructure();
     } else {
       renderError();
     }
   }
 
+  /**
+   * Renders a LocalService's Config shape in config.py.
+   * LocalService Config shapes are special:
+   * - They extend the default Smithy-Python Config
+   * - Their __init__ methods call super.__init__() to init Smithy-Python Config.
+   * Most of this is lifted directly from Smithy-Python; the changed components are
+   * called out with comments saying "Component below is changed from Smithy-Python."
+   */
+  protected void renderLocalServiceConfigShape() {
+    writer.addStdlibImport("typing", "Dict");
+    writer.addStdlibImport("typing", "Any");
+    var symbol = symbolProvider.toSymbol(shape);
+    // Component below is changed from Smithy-Python.
+    // Write special class that extends parent class.
+    writer.openBlock(
+      "class $L(Config):",
+      "",
+      symbol.getName(),
+      () -> {
+        writeProperties(false);
+        // Component below is changed from Smithy-Python.
+        // Write special __init__ that initializes parent class.
+        writeLocalServiceConfigShapeInit();
+        writeAsDict(false);
+        writeFromDict(false);
+        writeRepr(false);
+        writeEq(false);
+      }
+    );
+    writer.write("");
+  }
+
+  /**
+   * Writes an __init__ method for a LocalService Config shape.
+   * Most of this is lifted directly from Smithy-Python; the changed components are
+   * called out with comments saying "Component below is changed from Smithy-Python."
+   */
+  protected void writeLocalServiceConfigShapeInit() {
+    writer.openBlock(
+      "def __init__(",
+      "):",
+      () -> {
+        writer.write("self,");
+        if (!shape.members().isEmpty()) {
+          // Adding this star to the front prevents the use of positional arguments.
```

Comment:
Was this lifted from smithy-python too then? It's a great idea but I wasn't sure if they'd acted on it yet.

Reply:
Yes